routing/endpointregistry.go

```diff
@@ -53,26 +91,23 @@ func (r *EndpointRegistry) Do(routes []*Route) []*Route {
 	for _, route := range routes {
 		if route.BackendType == eskip.LBBackend {
 			for _, epi := range route.LBEndpoints {
-				metrics := r.GetMetrics(epi.Host)
-				if metrics.DetectedTime().IsZero() {
-					r.SetDetectedTime(epi.Host, now)
-				}
```

Comment:
Lets revert this logic and make SetDetected a simple setter (i.e. allow overwrite)

Reply:
How can we be sure there will be no races? If compare and set are not atomic, then in theory detected time set to this field could be later than the actual (first) detected time.

Reply:
Discussed with Alexander, decided to revert this logic because:
+ it's better not to overcomplicate endpointregistry interface without reason
+ there are no concurrent callers setting detected time at this point